packages/lib/src/gfx/shaders/UberMaterial.js

```diff
@@ -197,187 +121,257 @@ UberMaterial.prototype.uberOptions = {
   },
 };
 
-UberMaterial.prototype.copy = function (source) {
-  THREE.RawShaderMaterial.prototype.copy.call(this, source);
-
-  this.fragmentShader = source.fragmentShader;
-  this.vertexShader = source.vertexShader;
-
-  this.uniforms = THREE.UniformsUtils.clone(source.uniforms);
-  this.defines = { ...source.defines };
-  this.extensions = source.extensions;
+class UberMaterial extends THREE.RawShaderMaterial {
+  constructor(params) {
+    super(params);
+    // add fog
+    this.fog = true;
+    // used for instanced geometry
+    this.instancedPos = false;
+    this.instancedMatrix = false;
+    // atoms and links color
+    this.attrColor = false;
+    // second link color for cylinders
+    this.attrColor2 = false;
+    //
+    this.attrAlphaColor = false;
+    // overrides color for all vertices (used in selection)
+    this.overrideColor = false;
+    // zsrpites
+    this.sphereSprite = false;
+    this.cylinderSprite = false;
+    // clip Surfs individually
+    this.zClip = false;
+    // clip scene with global clip plane
+    this.clipPlane = false;
+    // enable fake (chess-like) opacity
+    this.fakeOpacity = false;
+    // render only depth, don't take care about the pixel color (used for transparency depth prepass)
+    this.prepassTransparancy = false;
+    // used to render pixel positions
+    this.colorFromPos = false;
+    // used to render shadowmap
+    this.shadowmap = false;
+    // used to describe shadowmap type
+    this.shadowmapType = 'random';
+    // used to render pixel view deph
+    this.colorFromDepth = false;
+    // mark that rendering is for orthographic camera
+    this.orthoCam = false;
+    // used to render dashed line
+    this.dashedLine = false;
+    // mark as transparent
+    this.transparent = true;
+    // mark as thick lines
+    this.thickLine = false;
+    // makes fog begin transparency (required for transparent background)
+    this.fogTransparent = false;
+    // used to render surface normals to G buffer for ssao effect
+    this.normalsToGBuffer = false;
+    // used for toon material
+    this.toonShading = false;
+    // set default values
+    this.setValues.call(this, {
+      uniforms: THREE.UniformsUtils.clone(defaultUniforms),
+      vertexShader: this.precisionString() + vertexShader,
+      fragmentShader: this.precisionString() + fragmentShader,
+      lights: true,
+      fog: true,
+      side: THREE.DoubleSide,
+    });
+    this.uberOptions = uberOptions;
+    this.setValues(params);
```

Comment:
can we set all values with a single setValues call?

Reply:
Probably yes, we can